### PR TITLE
Use scanelf over ldconfig on musl-libc distros

### PR DESCRIPTION
**Pull request by am11**
`ldconfig` on muscle libc based distros yields no results. The alternative is to use `scanelf`. For example, Julia platform package in aports uses `scanelf` to capture their expected results equivalent to `ldconfig`: https://github.com/alpinelinux/aports/blob/0042419fcc/unmaintained/julia/find-syslibs.

Since corefx produces portable linux-musl libc packages, this pr also relaxes the platform == alpine condition to instead introspect musl-libc, in order to support other distros like Void Linux (musl edition).

### CODE REVIEW

**Comment by am11** [comment]
To validate these changes, I used:

Alpine

```sh
docker run -it alpine

apk add curl bash libunwind icu-libs libintl

curl -sSL https://raw.githubusercontent.com/am11/cli/scanelf/scripts/obtain/dotnet-install.sh | bash /dev/stdin -Channel release/3.0.1xx

~/.dotnet/dotnet new console -o /tmp/testdrive
~/.dotnet/dotnet run -p /tmp/testdrive

# Hello World!
```

Void Linux (musl-libc edition)

```sh
docker run -it voidlinux/voidlinux-musl dash   # note: without "-musl" is the glibc edition

# inside the container
xbps-install -S
xbps-install -y curl bash ncurses pax-utils libunwind gettext icu-libs

curl -sSL https://raw.githubusercontent.com/am11/cli/scanelf/scripts/obtain/dotnet-install.sh | bash /dev/stdin -Channel release/3.0.1xx

export CLR_OPENSSL_VERSION_OVERRIDE='46' # because distro provides libressl

~/.dotnet/dotnet new console -o /tmp/testdrive
~/.dotnet/dotnet run -p /tmp/testdrive

# Hello World!
```

File: scripts/obtain/dotnet-install.sh [hunk 1]
```diff
@@ -173,10 +182,10 @@ get_current_os_name() {
         local linux_platform_name
         linux_platform_name="$(get_linux_platform_name)" || { echo "linux" && return 0 ; }
 
-        if [[ $linux_platform_name == "rhel.6" ]]; then
+        if [ "$linux_platform_name" = "rhel.6" ]; then
```
Comment (johnbeisner) [inline scripts/obtain/dotnet-install.sh]: What is the purpose of this change?

Comment (am11) [reply scripts/obtain/dotnet-install.sh]: It was to improve readability of this `function get_current_os_name()`; rest of `if` conditions in this function (and most in this file) are using POSIX shell's syntax for comparing strings (single square brackets with proper quotes).

**Comment by am11** [comment]
@livarcocc, can I get a review?

**Comment by livarcocc** [comment]
@johnbeisner will review when he is back in the office.

**Comment by am11** [comment]
@johnbeisner, saw you are back. Could you please review? :)

File: scripts/obtain/dotnet-install.sh [hunk 2]
```diff
@@ -159,6 +164,10 @@ get_linux_platform_name() {
     return 1
 }
 
+is_musl_based_distro() {
+    (ldd 2>&1 || true) | grep -q musl
```
Comment (johnbeisner) [inline scripts/obtain/dotnet-install.sh]: Can you explain how running "ldd" determines weather a machine is musl based - ldd requires a parameter on a Ubuntu machine. Have you tested this on Alpine? The parsing for "alpine" "void" "gentoo" in the platform name seems safer.

File: scripts/obtain/dotnet-install.sh [hunk 3]
```diff
@@ -245,16 +259,24 @@ check_pre_reqs() {
...
-        if [ ! -x "$(command -v ldconfig)" ]; then
-            echo "ldconfig is not in PATH, trying /sbin/ldconfig."
-            LDCONFIG_COMMAND="/sbin/ldconfig"
+        if is_musl_based_distro; then
+            if ! command -v scanelf > /dev/null; then
+                say_warning "scanelf not found, please install pax-utils package."
+                return 0
+            fi
+            LDCONFIG_COMMAND="scanelf --ldpath -BF '%f'"
+            [ -z "$($LDCONFIG_COMMAND 2>/dev/null | grep libintl)" ] && say_warning "Unable to locate libintl. Probable prerequisite missing; install libintl (or gettext)."
```
Comment (johnbeisner) [inline scripts/obtain/dotnet-install.sh]: Is 'libintl' required to run the .NET Core platform on all musl variants; the majority of the variants?

File: scripts/obtain/dotnet-install.sh (hunk 2 above)
Comment (am11) [reply scripts/obtain/dotnet-install.sh]: `ldd` without arguments on musl-libc distributions outputs:

```
musl libc (x86_64)
Version 1.1.22
Dynamic Program Loader
Usage: /lib/ld-musl-x86_64.so.1 [options] [--] pathname
```

Generally, to my knowledge, the most portable way to determine libc flavor (glibc, llvm-libc, ulibc, musl-libc etc.) on Unix-like OS is: `ldd --version 2>&1`. Alpine Linux' official package repo also has such example: https://github.com/alpinelinux/aports/blob/bfc00db0e316c32ea41ee272621ca1a24a1fde5e/community/openjdk10/build.patch#L22.

File: scripts/obtain/dotnet-install.sh (hunk 3 above)
Comment (am11) [reply scripts/obtain/dotnet-install.sh]: In the base system or minimal install of Alpine Linux and Void Linux, `libintl` is not installed. If we successfully install dotnet, and run the first command (e.g. `dotnet new`), we get:

```
Failed to load , error: Error loading shared library libintl.so.8: No such file or directory (needed by /root/.dotnet/shared/Microsoft.NETCore.App/3.0.0/libcoreclr.so)
Failed to bind to CoreCLR at '/root/.dotnet/shared/Microsoft.NETCore.App/3.0.0/'
Failed to create CoreCLR, HRESULT: 0x80008088
```

With this addition, user gets install-time error about the missing dependency (like this script already has for libunwind etc.). If some other musl-libc based distro or full version of Alpine/Void already has this lib installed, this condition will be skipped.

**Comment by johnbeisner** [comment]
@am11 
Thank you for your contribution.
